src/Analyser/MutatingScope.php

```diff
@@ -1051,12 +1051,24 @@ private function resolveType(Expr $node): Type
 			} else {
 				$right = $node->right;
 			}
+			$rightType = $this->getType($right);
+
+			$stringType = $rightType->toString();
+			if (
+				$node instanceof Node\Expr\BinaryOp\Mod
+				|| $node instanceof Node\Expr\AssignOp\Mod
+			) {
+				if ($stringType instanceof ConstantScalarType && $stringType->getValue() === '1') {
+					return new ConstantIntegerType(0);
+				}
+			}
+
+			$rightScalarTypes = TypeUtils::getConstantScalars($rightType->toNumber());
+			foreach ($rightScalarTypes as $scalarType) {
 
-			$rightTypes = TypeUtils::getConstantScalars($this->getType($right)->toNumber());
-			foreach ($rightTypes as $rightType) {
 				if (
-					$rightType->getValue() === 0
-					|| $rightType->getValue() === 0.0
+					$scalarType->getValue() === 0
+					|| $scalarType->getValue() === 0.0
```

Comment:
What about `'0'` here? Again, you can try `->toInteger()`.

Reply:
this code was existant before this PR and covers a different use-case.
should I change it with this PR nevertheless , or better do so in a different PR?

Reply:
You can do it here :)

Reply:
newly added testcases in 578e96d show that division by '0' and other falsy values work as expected.
so IMO we just leave this as is, right?